src/org/zaproxy/zap/extension/fieldenumeration/FieldEnumeration.java

```diff
@@ -0,0 +1,661 @@
+        	return;
+    		}
+ 
+    		try {
+        		Session session = Model.getSingleton().getSession();
+        		HistoryReference ref = new HistoryReference(session, HistoryReference.TYPE_ZAP_USER, httpMessage);
+        		final ExtensionHistory extHistory = Control.getSingleton().getExtensionLoader().getExtension(ExtensionHistory.class);
+        			if (extHistory != null) {
+            				extHistory.addHistory(ref);
+        			}
+        		SessionStructure.addPath(Model.getSingleton().getSession(), ref, httpMessage);
+    		} catch (HttpMalformedHeaderException | DatabaseException e) {
+        		LOGGER.warn("Failed to persist message sent:", e);
+    		}
+	}
+	 
+    	public List<NameValuePair> getParamList() {
+        	return listParam;
+    	}
+
+
+	protected String getEscapedName(HttpMessage msg, String name) {
+        	return name != null ? AbstractPlugin.getURLEncode(name) : "";
+    	}
+
+	public String setParameter(HttpMessage msg, NameValuePair originalPair, String name, String value) {
+        	return this.setParameter(msg, originalPair, name, value, false);
+	}
+
+    	public String setEscapedParameter(HttpMessage msg, NameValuePair originalPair, String name, String value) {
+        	return this.setParameter(msg, originalPair, name, value, true);
+    	}
+
+	private String setParameter(HttpMessage msg, NameValuePair originalPair, String name, String value, boolean escaped) {
+		ParameterParser parser;
+		parser = Model.getSingleton().getSession().getFormParamParser(msg.getRequestHeader().getURI().toString());
+		StringBuilder sb = new StringBuilder();
+		String encodedValue = new String();
+
+		try {
+	        	encodedValue = (escaped) ? value : URLEncoder.encode(value, StandardCharsets.UTF_8.name());
+		} catch (UnsupportedEncodingException ignore) {
+		}
+
+		System.out.println("Encoder " + encodedValue);
+        	NameValuePair pair;
+        	boolean isAppended = true;
+
+        	for (int i = 0; i < getParamList().size(); i++) {
+            		pair = getParamList().get(i);
+			
+			if (pair.getName() == name) {
+				isAppended = paramAppend(sb, getEscapedName(msg, name), encodedValue, parser);
+			}
+			else {
+				try {
+       	   				isAppended = paramAppend(sb, getEscapedName(msg, pair.getName()), URLEncoder.encode(pair.getValue(), StandardCharsets.UTF_8.name()), parser);
+				} catch (UnsupportedEncodingException ignore) {
+				}
+			}
+
+            		if (isAppended && i < getParamList().size() - 1) {
+                		sb.append(parser.getDefaultKeyValuePairSeparator());
+            		}
+        	}
+
+        	if (sb.length() == 0) {
+             //No original query string
+            		sb.append(encodedValue);
+        	}
+        	String query = sb.toString();
+		System.out.println("Query " + query);
+		msg.getRequestBody().setBody(query);
+        	return query;
+
+	}
+
+	private boolean paramAppend(StringBuilder sb, String name, String value, ParameterParser parser) {
+       		boolean isEdited = false;
```

Comment:
This seems to be intended one level too far..

Reply:
meaning?

Reply:
Meaning I can't spell....sorry.

Should have said "...indented one level too far...."